controllers/loadtest_controller_test.go

```diff
@@ -502,3 +501,134 @@ var _ = Describe("Pod Creation", func() {
 		})
 	})
 })
+
+var _ = Describe("CheckMissingPods", func() {
+
+	var currentLoadTest = newLoadTestWithMultipleClientsAndServers() // nothing modify this --> okay
+	var expectedReturnList []*grpcv1.Component                       //initialized every time after a test -->okay
+	var allRunningPods = &corev1.PodList{Items: []corev1.Pod{}}
+	var returnedList []*grpcv1.Component // it was a newlist generated form myfun everytime, address are different --> okay
+
+	JustBeforeEach(func() {
+		returnedList = CheckMissingPods(currentLoadTest, allRunningPods)
+	})
+
+	AfterEach(func() {
+		expectedReturnList = []*grpcv1.Component{}
+		newThing := corev1.PodList{Items: []corev1.Pod{}}
+		allRunningPods = &newThing
+	})
```

Comment:
The slice declaration in line 508 is fine, because this is a nil slice. It is not actually initialized until you `amend` to it. I would prefer if the assignment in lines 507 and 509 are moved into a `BeforeEach`. I see your logic and its valid for the current state of the code. But, imagine in the future that Paulo or I come add an `It` block. We modify `currentLoadTest`. Then, we're not hermetic. It is better to put all initializations in a `BeforeEach`.

Prefer a fresh initialization with a `BeforeEach` over a clean-up with `AfterEach`. Clean-ups are much harder to get right, because each test must rely on the successful clean-up of the previous one.

Other than this, this section LGTM.

Reply:
Agreed for the concept of getting a fresh start other than have next test depend on previous clean up. Re-construct the test structure to reflect the change.